pkg/asset/installconfig/aws/validation.go

```diff
@@ -75,6 +72,53 @@ func validatePlatform(ctx context.Context, meta *Metadata, fldPath *field.Path,
 	return allErrs
 }
 
+func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorList {
+	// accept AMI from the rhcos stream metadata
+	if sets.NewString(rhcos.AMIRegions...).Has(config.Platform.AWS.Region) {
+		return nil
+	}
+
+	// accept AMI specified at the platform level
+	if config.Platform.AWS.AMIID != "" {
+		return nil
+	}
+
+	// accept AMI specified for the default machine platform
+	if config.Platform.AWS.DefaultMachinePlatform != nil {
+		if config.Platform.AWS.DefaultMachinePlatform.AMIID != "" {
+			return nil
+		}
+	}
+
+	// accept AMIs specified specifically for each machine pool
+	controlPlaneHasAMISpecified := false
+	if config.ControlPlane != nil && config.ControlPlane.Platform.AWS != nil {
+		controlPlaneHasAMISpecified = config.ControlPlane.Platform.AWS.AMIID != ""
+	}
+	computesHaveAMISpecified := true
+	for _, c := range config.Compute {
+		if c.Replicas != nil && *c.Replicas == 0 {
+			continue
+		}
+		if c.Platform.AWS != nil && c.Platform.AWS.AMIID == "" {
```

Comment:
Should this be `c.Platform.AWS == nil ||`?

Reply:
No, I don't think so. The test here is to find if there is a compute machine pool which explicitly has an AMI specified. If `c.Platform.AWS` is nil, then the compute machine pool does not have an AMI specified.

Reply:
Looking at this again, you are correct. The logic should be as you stated instead of what I have here. I will make the change.